src/core/server/http/http_server.ts

```diff
@@ -127,4 +138,35 @@ export class HttpServer {
     const routePathStartIndex = routerPath.endsWith('/') && routePath.startsWith('/') ? 1 : 0;
     return `${routerPath}${routePath.slice(routePathStartIndex)}`;
   }
+
+  private registerOnRequest = (fn: OnRequest) => {
+    if (this.server === undefined) {
+      throw new Error('Server is not created yet');
+    }
+
+    this.server.ext('onRequest', adoptToHapiOnRequestFormat(fn));
+  };
+
+  private registerAuth = async <T>(fn: Authenticate<T>, cookieOptions: CookieOptions<T>) => {
```

Comment:
__suggestion:___ `cookieOptions` vs `sessionStorage` --- we should probably tweak naming a bit so that there is a clear link between "encrypted cookie based state storage options" we pass with the auth handler and the "encrypted cookie based state storage instance" we receive as the second handler parameter.